### PR TITLE
Remove the deprecated addToBody option and return files as buffers when attachFieldsToBody is set to keyValues

### LINKED ISSUE #202: increase code coverage
as titled

### LINKED ISSUE #289: migrate out of deprecated API.
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

3.x.x

### Plugin version

_No response_

### Node.js version

14.x

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

20.04

### Description

Migrate out of deprecated API.

https://github.com/fastify/fastify-multipart/blob/37165757947857cd8b3977ba362c4536fd2d837e/index.js#L36-L44

`request.multipart` is actually deprecated, it should not be used inside this plugin.

Reported by Jones on Discord Server.

### Steps to Reproduce

Register the plugin with `addToBody`.
```js
fastify.register(multipart, {
  addToBody: true,
});
```

### Expected Behavior

No deprecation warning.

### LINKED ISSUE #435: Limit and File Handling on addToBody Option
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.15.0

### Plugin version

7.6.0

### Node.js version

16.13

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

13.3.1

### Description

When using `@fastify/multipart` empty files are returned when the file size limit is exceeded (default: 1mb).

If I send files larger than the limit to the API and access them body, I receive empty files when the file size exceeds the limit.
e.g:

File with 1.1mb
`{
  data: undefined,
  filename: 'DJI_0366.JPG',
  encoding: '7bit',
  mimetype: 'image/jpeg',
  limit: true
}`

If the file is smaller then the limit everything works correctly and the data field is defined.

File with 0.8mb:

`{
  data: <Buffer ff d8 ff e0 00 10 4a 46 49 46 00 01 01 00 00 01 00 01 00 00 ff e2 02 28 49 43 43 5f 50 52 4f 46 49 4c 45 00 01 01 00 00 02 18 00 00 00 00 04 30 00 00 ... 212574 more bytes>,
  filename: 'DJI_0365.JPG',
  encoding: '7bit',
  mimetype: 'image/jpeg',
  limit: false
}`
### Reason

```
file.on('limit', () => {
    lastFile.limit = true
  })
file.on('end', () => {
    if (!lastFile.limit) {
      lastFile.data = Buffer.concat(fileData)
    } else {
      lastFile.data = undefined
    }
})
```

### Steps to Reproduce

Register module with the `addToBody` Option 

` app.register(fastifyMultipart, {
    addToBody: true
  });
  `
  
  And access the files e.g. using nestjs
  `  @Post()
  post(@Body() body: { files: Buffer[] }): void {
    return this.appService.createFiles(body.files);
  }`

### Expected Behavior

Either ban this option or return a 413 error message if the file exceeds the size limit. 
This is a serious issue that can result in data loss

### LINKED ISSUE #447: onFile hook doesn't work on files array field with (attachFieldsToBody: "keyValues")
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.17.0

### Plugin version

7.6.0

### Node.js version

18.15.0

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

13.2 (Ventura)

### Description

I want to attach files to my body which also contains other data like strings and booleans.

I used the **attachFieldsToBody: "keyValues"** option, and it seems to work.

**There are however two problems:**

1) the file is attached to the body with the default value buffer.toString(). Here it doesn't appear to convert to a base64 or utf-8 string due to the presence of unknown characters. So it's impossible to restore the buffer and therefore the file from that string. I managed to get around this by using the **onFile hook** which allowed me to return a correct base64 string.

```
onFile: async function (part) {
  const buff = await part.toBuffer();
  const base64_file = buff.toString("base64");
  const type = await fileTypeFromBuffer(buff);
  part.value = {
      file: base64_file,
      type: type,
  };
}
```

2) The problem now is that if I pass more than one file into the same form field, I get returned an array of strings that are not "processed" by the "onFile" hook and are therefore unusable. it's like the onFile hook only works if there is a single file in a specific key of the body, if it's an array it doesn't get executed.

Can anyone come and help me?

### Steps to Reproduce

**1) package.json**

```
{
  "name": "fastifyplayground",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@fastify/multipart": "^7.6.0",
    "fastify": "^4.17.0",
    "file-type": "^18.5.0",
    "mime-types": "^2.1.35"
  },
  "type": "module"
}
```

**2) index.js**

```
import Fastify from "fastify";
import { fileTypeFromBuffer } from "file-type";

// Require the framework and instantiate it
const fastify = Fastify({
    logger: true,
});

// register multipart plugin
fastify.register(import("@fastify/multipart"), {
    attachFieldsToBody: "keyValues",
    limits: {
        fileSize: 1048576, // 1MB
    },
    onFile: async function (part) {
        const buff = await part.toBuffer();
        const base64_file = buff.toString("base64");
        const type = await fileTypeFromBuffer(buff);
        part.value = {
            file: base64_file,
            type: type,
        };
    },
});

// Declare a route
fastify.route({
    method: "POST",
    url: "/upload",
    handler: async (request, reply) => {
        // get body values
        const test = request.body.test;
        const image = request.body.image;

        // response
        return {
            test: test,
            image: image,
        };
    },
});

// Run the server!
const start = async () => {
    try {
        await fastify.listen({ port: 3000 });
    } catch (err) {
        fastify.log.error(err);
        process.exit(1);
    }
};
start();

```

### Expected Behavior

_No response_

### PR DESCRIPTION
**Pull request by gurgunday**
There has been [multiple](https://github.com/fastify/fastify-multipart/issues/447#issuecomment-1704377333) [requests](https://github.com/fastify/fastify-multipart/issues/447#issuecomment-1734334541) to make a major release as we fixed an annoying issue, but so far we've held it back to combine these changes with a rewrite of the plugin

I think as @mcollina said, bumping major is cheap and we can release v9 in the future when the rewrite is complete

Closes #447 and closes #456, closes #289, closes #202 closes #435

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by Eomm** [comment]
Does it close this issue too https://github.com/fastify/fastify-multipart/issues/464?

**Comment by gurgunday** [comment]
That one is more about the actual rewrite we wanted to do. Maybe we can remove “restructure” from its title

The following however can be closed as we removed the cause of the bug while providing an alternative that works: https://github.com/fastify/fastify-multipart/issues/435

File: test/multipart-attach-body.test.js [hunk 1]
```diff
@@ -434,3 +439,50 @@ test('should handle file stream consumption when internal buffer is not yet load
   await once(res, 'end')
   t.pass('res ended successfully')
 })
+
+test('should pass the buffer instead of convering to string', async function (t) {
```
Comment (Fdawgs) [inline test/multipart-attach-body.test.js:443]:
```suggestion
test('should pass the buffer instead of converting to string', async function (t) {
```

**Comment by gurgunday** [comment]
Anyone wants to add a breaking change?

Last call!

File: README.md [hunk 2]
```diff
@@ -240,13 +238,13 @@ fastify.post('/upload/files', async function (req, reply) {
 })
 ```
 
-Request body key-value pairs can be assigned directly using `attachFieldsToBody: 'keyValues'`. Field values will be attached directly to the body object. By default, all files are converted to a string using `buffer.toString()` used as the value attached to the body.
+Request body key-value pairs can be assigned directly using `attachFieldsToBody: 'keyValues'`. Field values including file buffers will be attached to the body object.
```
Comment (Fdawgs) [inline README.md:241]:
```suggestion
Request body key-value pairs can be assigned directly using `attachFieldsToBody: 'keyValues'`. Field values, including file buffers, will be attached to the body object.
```

**Review by Fdawgs — CHANGES_REQUESTED** [review]

**Comment by Fdawgs** [comment]
Probably also needs a more descriptive PR title rather than v8? 😋

**Review by Fdawgs — APPROVED** [review]

**Review by Uzlopak — APPROVED** [review]
If @Fdawgs approves, I also approve.

**Pull request merged**